server/model/Game_manager.c

```diff
@@ -116,6 +136,23 @@ void		set_game_status(uint game_status)
   (*game_info)->game_status = game_status;
 }
 
+void		energy_fall(uint map_size)
+{
+  int		x;
+  int		y;
+  int		power;
+  t_game_info	**game_info;
+
+  game_info = get_info();
+  srand(time(NULL));
+  x = rand() % (map_size + 1);
+  y = rand() % (map_size + 1);
+  power = rand() % (15 - 5) + 5;
+  if (add_link(&((*game_info)->energy_cells), create_energy_cell(x, y, power)))
```

Comment:
Je pense qu'il est préférable d'appeller create_energy_cell avant et de tester sur NULL.
En l'état, si create_energy_cell n'arrive pas à malloc, elle enverra NULL à add_link.
add_link ne bronchera pas, j'avais fais le choix de ne pas gérer le content dans add_link pour laisser l'utilisateur mettre ce qu'il veut (si il veut un maillon avec un content à NULL, c'est son problème) mais du coups, pas de retour d'erreur de la part d'add_link.
Donc, si tu envois NULL dans add_link, à un moment, tu vas itérer sur les maillons, sortir le content et le caster dans une t_energy_cell et bim => segfault très chiant à tracker!
Je vais modifier add_link je pense, mais en attendant, un ptit test sur create_energy_cell serait plus safe!

Reply:
ouaip c'était prévu :) je voulais juste boucler la fonction avant d'arrêter de taffer dessus ^^ toutes mes fonctions ne sont pas complétes et j'ai vu qu'on gérait pas encore beaucoup de choses sur les energy cells :) j'vais essayer d'améliorer ça !